version: "3.8"

networks:
  selenium-network:
    driver: bridge

services:
  docker-mysql:
    image: mysql:8.0
    restart: always
    container_name: mysql-db
    environment:
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
      MYSQL_DATABASE: ${MYSQL_DATABASE}
    ports:
      - "3306:3306"
    volumes:
      - ./db/:/var/lib/mysql
    networks:
      - selenium-network
    env_file:
      - .env
    healthcheck:
      test:
        [
          "CMD",
          "mysqladmin",
          "ping",
          "-h",
          "localhost",
          "-u",
          "root",
          "-p${MYSQL_ROOT_PASSWORD}",
        ]
      interval: 5s
      timeout: 5s
      retries: 20

  app:
    build:
      context: .
      dockerfile: Dockerfile
    depends_on:
      docker-mysql:
        condition: service_healthy
    restart: always
    env_file:
      - .env
    ports:
      - "8080:8080"
    environment:
      SPRING_DATASOURCE_URL: jdbc:mysql://docker-mysql:3306/${MYSQL_DATABASE}
      SPRING_DATASOURCE_USERNAME: root
      SPRING_DATASOURCE_PASSWORD: ${MYSQL_ROOT_PASSWORD}
    networks:
      - selenium-network

  grafana:
    image: grafana/grafana:latest
    container_name: grafana
    depends_on:
      - app
    environment:
      GF_SECURITY_ADMIN_PASSWORD: ${GF_SECURITY_ADMIN_PASSWORD}
      GF_PROVISIONING_PATH: /etc/grafana/provisioning
    ports:
      - "3000:3000" # Expose Grafana on port 3000
    networks:
      - selenium-network
    env_file:
      - .env
    volumes:
      - ./grafana-data:/var/lib/grafana
      - ./grafana/provisioning/datasources:/etc/grafana/provisioning/datasources
      - ./grafana/provisioning/dashboards:/etc/grafana/provisioning/dashboards
      - ./grafana/dashboards:/var/lib/grafana/dashboards

volumes:
  db:
  grafana-data:
